index.js

```diff
@@ -46,15 +49,21 @@ function ReposUpload(config) {
   }
 
   function createFile(fileUrl, data, callback) {
+    var versionedProps = {};
+    if (typeof callback === 'object' && typeof arguments[3] === 'function') {
+      versionedProps = callback;
+      callback = arguments[3];
```

Comment:
I'd say this is the common way to support varying number of input arguments in Javascript :)

Just as a side-note ES6 introduces optional arguments however.